File: sdk/python/examples/adk/00_hello_world.py
# Copyright (c) 2025 Agentspan
# Licensed under the MIT License. See LICENSE file in the project root for details.

"""Minimal Google ADK greeting agent — for debugging the native runner.

The simplest possible ADK agent: no tools, no structured output, one turn.
Used to verify the ADK native shim works end-to-end before testing more
complex examples.

Requirements:
    - pip install google-adk
    - GOOGLE_API_KEY or GEMINI_API_KEY environment variable
    - AGENTSPAN_LLM_MODEL=google_gemini/gemini-2.0-flash (for AgentSpan runs)
    - AGENTSPAN_SERVER_URL=http://localhost:8080/api (for AgentSpan runs)
"""

from google.adk.agents import Agent

from agentspan.agents import AgentRuntime

from settings import settings

agent = Agent(
    name="greeter",
    model=settings.llm_model,
    instruction="You are a friendly greeter. Reply with a warm hello and one fun fact.",
)

with AgentRuntime() as runtime:
    result = runtime.run(agent, "Say hello!")
    result.print_result()

File: sdk/typescript/examples/adk/00-hello-world.ts
/**
 * Minimal Google ADK Greeting Agent.
 *
 * The simplest possible ADK agent: no tools, no structured output, one turn.
 * Used to verify the ADK integration works end-to-end.
 *
 * Requirements:
 *   - npm install @google/adk zod
 *   - AGENTSPAN_SERVER_URL for agentspan path
 */

import { LlmAgent } from '@google/adk';
import { AgentRuntime } from '../../src/index.js';

const model = process.env.AGENTSPAN_LLM_MODEL ?? 'gemini-2.5-flash';

const agent = new LlmAgent({
  name: 'greeter',
  model,
  instruction: 'You are a friendly greeter. Reply with a warm hello and one fun fact.',
});

// ── Run on agentspan ───────────────────────────────────────────────

async function main() {
  const runtime = new AgentRuntime();
  try {
    const result = await runtime.run(agent, 'Say hello!');
    console.log('Status:', result.status);
    result.printResult();
  } finally {
    await runtime.shutdown();
  }
}

main().catch(console.error);
